src/uct/ib/base/ib_iface.c

```diff
@@ -550,43 +560,47 @@ uct_ib_iface_res_domain_init(uct_ib_iface_res_domain_t *res_domain,
     uct_ib_device_t *dev = uct_ib_iface_device(iface);
     struct ibv_exp_res_domain_init_attr attr;
 
-    attr.comp_mask    = IBV_EXP_RES_DOMAIN_THREAD_MODEL |
-                        IBV_EXP_RES_DOMAIN_MSG_MODEL;
-    attr.msg_model    = IBV_EXP_MSG_LOW_LATENCY;
-
-    switch (iface->super.worker->thread_mode) {
-    case UCS_THREAD_MODE_SINGLE:
-        attr.thread_model = IBV_EXP_THREAD_SINGLE;
-        break;
-    case UCS_THREAD_MODE_SERIALIZED:
-        attr.thread_model = IBV_EXP_THREAD_UNSAFE;
-        break;
-    default:
-        attr.thread_model = IBV_EXP_THREAD_SAFE;
-        break;
-    }
+    if (iface->config.res_domain) {
```

Comment:
instead, we can call uct_worker_tl_data_get() only if res_domain==1, in ib_iface.c:667
```
if (res_domain_key == UCT_IB_IFACE_NULL_RES_DOMAIN_KEY || !res_domain) {
  self->res_domain = NULL;
} else {
...
```

Reply:
done